src/components/generics/vacation-test-forms/my-vacation-requests.tsx

```diff
@@ -65,6 +71,70 @@ const RenderVacationRequests = () => {
     initializeRequests();
   }, [person]);
 
+  /**
+   * Initializes all vacation request statuses
+   */
+  const initializeRequestStatuses = async () => {
```

Comment:
It seems there is some duplication here from the previous file, can we reuse the same function rather than have it in 2 places? IF this is not possible then please update this function according to the comments for the other function e.g. avoid use of type assertions for the request.id!

Reply:
decided to leave the duplicates in, made it check if request.id exists or not